Sources/tuist/tuist.docc/Articles/Tuist/Users/dependencies.md

```diff
@@ -4,11 +4,167 @@ Learn how to declare internal and external dependencies in your project.
+- `XCTest`: Declares a dependency with XCTest.
+
+Note that every dependency type accepts a `condition` option to conditionally link the dependency based on the platform. By default, it links the dependency for all platforms the target supports.
+
+> Warning: We haven't yet solved the problem of targets being able to import dependencies that they shouldn't. Some users have implemented their custom solutions to detect this, but we haven't yet found a solution that we're happy with. We are currently exploring customizing the directory where products are outputted to solve this problem.
 
 ## External dependencies
 
+Tuist also allows you to declare external dependencies in your project.
+
+### Carthage
+
+Since [Carthage](https://github.com/carthage/carthage) outputs `frameworks` or `xcframeworks`, you can run `carthage update` to output the dependencies in the `Carthage/Build` directory and then use the `.framework` or `.xcframework` target dependency type to declare the dependency in your target. You can wrap this in a script that you can run before generating the project.
+
+```bash
+#!/usr/bin/env bash
+
+carthage update
+tuist generate
+```
+
+> Warning: Carthage dependencies are not compatible with workflows like `build` or `test` that run `xcodebuild` right after generating the project. We have plans to add support for pre and post generation hooks.
+
+### CocoaPods
+
+[CocoaPods](https://cocoapods.org) expects an Xcode project to integrate the dependencies. You can use Tuist to generate the project, and then run `pod install` to integrate the dependencies by creating a workspace that contains your project and the Pods dependencies. You can wrap this in a script that you can run before generating the project.
+
+```bash
+#!/usr/bin/env bash
+
+tuist generate
+pod install
+```
+
+> Warning: CocoaPods dependencies are not compatible with workflows like `build` or `test` that run `xcodebuild` right after generating the project. They are also incompatible with binary caching and selective testing since the fingerprinting logic doesn't account for the Pods dependencies.
+
 ### Swift Packages
 
+Swift Packages are our recommended way of declaring dependencies in your project.
+You can integrate them using Xcode's default integration mechanism or using Tuist's XcodeProj-based integration.
+
+#### Xcode's default integration
+
+If you want to use Xcode's default integration mechanism, you can pass the list `packages` when instantiating a project:
+
+```swift
+let project = Project(name: "MyProject", packages: [
+    .remote(url: "https://github.com/krzyzanowskim/CryptoSwift", requirement: .exact("1.8.0"))
+])
+```
+
+And then reference them from your targets:
+
+```swift
+let target = Target(name: "MyTarget", dependencies: [
+    .package(product: "CryptoSwift", type: .runtime)
+])
+```
+
+For Swift Macros and Build Tool Plugins, you'll need to use the types `.macro` and `.plugin` respectively.
+
+#### Tuist's XcodeProj-based integration
```

Comment:
I would keep this small and refer to the tutorial, WDYT?

Reply:
I think it's an amazing idea. I'll move it to a tutorial.

Reply:
There should already be one!